from_now/from_now.go

```diff
@@ -0,0 +1,112 @@
+package from_now
+
+import (
+	"fmt"
+	"github.com/taskcluster/taskcluster-cli/extpoints"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type from_now struct{}
+
+func init() {
+	extpoints.Register("from-now", from_now{})
+}
+
+func (from_now) ConfigOptions() map[string]extpoints.ConfigOption {
+	return nil
+}
+
+func (from_now) Summary() string {
+	return "Returns a timestamp which is DURATION ahead in the future."
+}
+
+func (from_now) Usage() string {
+	usage := "Usage: taskcluster from-now DURATION"
+	usage += "\n"
+	return usage
+}
+
+func (from_now) Execute(context extpoints.Context) bool {
+	duration := context.Arguments["DURATION"].(string)
+
+	offset, err := parseTime(duration)
+	
+	if err {
+		panic("String: '" + duration + "' isn't a time expression")
+	}
+
+	timeToAdd := time.Hour*time.Duration(offset.weeks*7*24) +
+		time.Hour*time.Duration(offset.days*24) +
+		time.Hour*time.Duration(offset.hours) +
+		time.Minute*time.Duration(offset.minutes) +
+		time.Second*time.Duration(offset.seconds)
+
+	timein := time.Now().Add(timeToAdd)
+	timein = timein.AddDate(offset.years, offset.months, 0)
+
+	fmt.Println(timein.Format(time.RFC3339))
+
+	return true
+}
+
+type parse_time struct {
+	years   int
+	months  int
+	weeks   int
+	days    int
+	hours   int
+	minutes int
+	seconds int
+}
+
+/* parseTime takes an argument `str` which is a string on the form `1 day 2 hours 3 minutes`
+ * where specification of day, hours and minutes is optional. You can also the
+ * short hand `1d2h3min`, it's fairly tolerant of different spelling forms and
+ * whitespace. But only really meant to be used with constants.
+ *
+ * Returns a parse_time object with all of the fields filled in with the correct values.
+ */
+func parseTime(str string) (parse_time, bool) {
```

Comment:
Does the `bool` return type indicate that the parsing failed? I would recommend using [Go's own error type](https://blog.golang.org/error-handling-and-go). You can do `return nil, errors.New("Your error message")` in case of failure.